tests/evaluator/test_text_classification.py

```diff
@@ -0,0 +1,73 @@
+import json
+import os
+import subprocess
+import unittest
+from datetime import datetime
+
+from datasets import load_dataset
+from transformers import AutoModelForSequenceClassification, AutoTokenizer, pipeline
+
+from evaluate import evaluator
+
+
+class TestTextClassificationEvaluator(unittest.TestCase):
+    def test_match_transformers_example(self):
+        model_name = "howey/bert-base-uncased-sst2"
+
+        tps = datetime.now().isoformat()
+        dir_path = f"/tmp/{tps}_textclassification"
+        if not os.path.exists(dir_path):
+            os.makedirs(dir_path)
+        os.chdir(dir_path)
+        subprocess.run(
+            "git clone --depth 3 --filter=blob:none --sparse https://github.com/huggingface/transformers", shell=True
+        )
+
+        os.chdir("transformers")
+        subprocess.run("git sparse-checkout set examples/pytorch/text-classification", shell=True)
+
+        subprocess.run(
+            f"python3 examples/pytorch/text-classification/run_glue.py"
+            f" --model_name_or_path {model_name}"
+            f" --task_name sst2"
+            f" --do_eval"
+            f" --max_seq_length 9999999999"  # rely on tokenizer.model_max_length for max_length
+            f" --output_dir {dir_path}/textclassification_sst2_transformers"
+            f" --max_eval_samples 200",
+            shell=True,
+        )
+
+        eval_filename = f"{dir_path}/textclassification_sst2_transformers/eval_results.json"
+
+        with open(eval_filename, "r") as f:
+            eval_dict = json.load(f)
```

Comment:
To make things a bit more compact could you change this to:

```suggestion
            transformer_results = json.load(f)
```

and later

```py
evaluator_results = e.compute(...)
```

so that at the end we can just run:

```py
self.assertEqual(transformer_results["eval_accuracy"], evaluator_accuracy["accuracy"])
```

Reply:
Done, this is nicer!